## Data were downloaded from GEO and analyzed using Partek implementation of R code. The code is shown for undifferentiated iPSCs and is the same as that used for human heart tissue data.

### Differential gene expression analysis for DEseq2 comparing AA vs AG samples:
`/opt/partek_flow/bin/tabular/flow_table --threads 0 -p /home/.../DESeq2Result-46744892665539348/nestedParameters4532302968282149753.json
/opt/partek_flow/bin/quantification/find_feature_indices --input /home/.../NormalizationResult-46600420919020999/normalized.mat.matrix --output /home/.../DESeq2Result-46744892665539348/filtered4872203228945901968.txt --threads 0 --any_missing --expression_method geometric mean --operation lt --expression_value 1.0
/opt/partek_flow/bin/datablock/filter_datablock --input /home/.../DESeq2Result-46744892665539348/input1838268785428157768.json --output /home/.../DESeq2Result-46744892665539348/output1217345651084406628.json --column-filter /home/.../DESeq2Result-46744892665539348/filtered4872203228945901968.txt --sort_indices --threads 0
/opt/partek_flow/bin/gene_analysis/flow_gsa --input_bdb /home/.../DESeq2Result-46744892665539348/input626723612015019970.json --model_config /home/.../DESeq2Result-46744892665539348/result/modelsComparisons.json --attributes /home/.../DESeq2Result-46744892665539348/parameters4712433874115524696.json --output_folder /home/.../DESeq2Result-46744892665539348/result --df 0 -c AICc --fdrstepup --level gene-level --deseq2 --indep_filter --ref_factor 97 --ref_level 643 --shrink_plots /home/.../DESeq2Result-46744892665539348/result/shrinkage_plots_info.txt --threads 0
/opt/partek_flow/bin/gene_analysis/flow_parseGSAResults --threads 0 /home/.../DESeq2Result-46744892665539348/result/gsa_result.txt /home/.../DESeq2Result-46744892665539348/result --multimodel
/opt/partek_flow/bin/annotation/filter_transcripts --threads 0 --input /home/.../ImportQuantificationResult-46600375796136502/Import.transcripts --gsa_result /home/.../DESeq2Result-46744892665539348/result/gsa_result.txt --output /home/.../DESeq2Result-46744892665539348/result/annotation.transcripts`

### Gene set enrichment analysis was performed for genes with p<0.05 using hallmark gene sets:
`/opt/partek_flow/bin/gene_ontology/gmt_filter --threads 0 --input /home/flow/FlowData/library_files_local/user3/hg38/Hallmark pathways/Gene set database/h.all.v7.5.1.symbols.gmt --output /home/.../GOEnrichmentResult-46744986059930518/filtered.gmt --report /home/.../GOEnrichmentResult-46744986059930518/gmt_filter_report.txt --annotation /home/.../ImportQuantificationResult-46600375796136502/Import.col.annotation
/opt/partek_flow/bin/gene_ontology/go_enrichment --threads 0 --input /home/.../GeneAnalysisFilterResult-46744946621252593/result/gsa_result.txt --gmt_file /home/.../GOEnrichmentResult-46744986059930518/filtered.gmt --output /home/.../GOEnrichmentResult-46744986059930518/gene_ontology_table.txt --report /home/.../GOEnrichmentResult-46744986059930518/report.txt --present_in_list /home/.../GOEnrichmentResult-46744986059930518/present.gmt --not_in_list /home/.../GOEnrichmentResult-46744986059930518/absent.gmt`

### Hierarchical clustering was performeded using Average linkage cluster distance metric and Euclidean point distance metric:
`/opt/partek_flow/bin/tabular/flow_table --threads 0 -p /home/.../HierarchicalClusteringResult-955060769312025/parameters7370262710846894185.json
/opt/partek_flow/bin/datablock/filter_datablock --input /home/.../HierarchicalClusteringResult-955060769312025/intermediate6651194116413231419.json --output /home/.../HierarchicalClusteringResult-955060769312025/intermediate8643939133961927666.json --row-vector-filter /home/.../HierarchicalClusteringResult-955060769312025/intermediate7154306345613803059.bin --threads 0
/opt/partek_flow/bin/datablock/normalize_datablock --input /home/.../HierarchicalClusteringResult-955060769312025/intermediate2723982944834538622.json --output /home/.../HierarchicalClusteringResult-955060769312025/intermediate38877774225229274.mat.matrix --normalization standardize --threads 0
/opt/partek_flow/bin/hierarchical_clustering/hcluster --input /home/.../HierarchicalClusteringResult-955060769312025/intermediate38877774225229274.mat.matrix --output /home/.../HierarchicalClusteringResult-955060769312025/featuresDendrogram.den --output-filter /home/.../HierarchicalClusteringResult-955060769312025/intermediate4139283403334118672.bin --data-dissimilarity euclidean --cluster-dissimilarity average_linkage --transpose --gpu --threads 0
/opt/partek_flow/bin/datablock/filter_datablock --input /home/.../HierarchicalClusteringResult-955060769312025/intermediate5449477435042387128.json --output /home/.../HierarchicalClusteringResult-955060769312025/intermediate3049857924561757707.json --column-vector-filter /home/.../HierarchicalClusteringResult-955060769312025/intermediate4139283403334118672.bin --threads 0
/opt/partek_flow/bin/datablock/filter_datablock --input /home/.../HierarchicalClusteringResult-955060769312025/intermediate8388493288674049638.json --output /home/.../HierarchicalClusteringResult-955060769312025/intermediate2444057646806332938.json --column-vector-filter /home/.../HierarchicalClusteringResult-955060769312025/intermediate4139283403334118672.bin --threads 0`


